tests/dht/test_providers.nim

```diff
@@ -35,34 +39,31 @@ proc initProvidersNode(
   let d = initDiscoveryNode(rng, privKey, address, bootstrapRecords)
   newProvidersProtocol(d)
 
-proc toPeerRecord(p: ProvidersProtocol) : PeerRecord =
-  ## hadle conversion between the two worlds
-
-  #NodeId is a keccak-256 hash created by keccak256.digest and stored as UInt256
-  let discNodeId = p.discovery.localNode.id
-  ## get it back to MDigest form
-  var digest: MDigest[256]
-  digest.data = discNodeId.toBytesBE
-  ## get into a MultiHash
-  var mh = MultiHash.init(multiCodec("keccak-256"), digest).orError(HashError).get()
-  result = PeerRecord.init(
-    peerId = PeerId.init(mh.data.buffer).get,
+proc toSignedPeerRecord(privKey: crypto.PrivateKey) : SignedPeerRecord =
+  ## handle conversion between the two worlds
+
+  let pr = PeerRecord.init(
+    peerId = PeerId.init(privKey.getPublicKey.get).get,
     seqNo = 0,
```

Comment:
seqNo is important to avoid "replay attacks", eg I publish an old SPR of yours, and you changed ip in between, so now you're unreachable
Libp2p recommends setting it to a timestamp of some sort, and rejecting any SPR with a seqNo <= the old one

Reply:
Removed `seqNo` in the `init` call as it defaults to `getTime().toUnix().uint64`.